File: top-interview-150/array/0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int ptr_1=0;
        int ptr_2=height.length-1;
        int max_area=0;
        while(ptr_1<ptr_2){
            if(height[ptr_1]<height[ptr_2]){
                max_area= Math.max(max_area,height[ptr_1]* (ptr_1 - ptr_2));
                ptr_1+=1;
            }else{
                max_area= Math.max(max_area,height[ptr_2]* (ptr_1 - ptr_2));
                ptr_2-=1;
            }
        }

        return max_area;
    }
}

File: tesla/array/0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {

    let n: number= height.length;
    let i : number= 0;
    let j : number= n-1;
    let area : number = 0;
    let max_area: number= 0;

    while(i<j){
        let width : number= j-i;
        let waterHeight= Math.min(height[i],height[j])

        area = width*waterHeight;

        max_area= Math.max(max_area,area);

        if(height[i]<height[j]){
            i++;
        }else{
            j--;
        }
    }

    return max_area;
};